main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -270,14 +284,46 @@ object Safety extends Phase[Root, Root] {
       case Polarity.Negative =>
         // Compute the free variables in the terms which are *not* bound by the lexical scope.
         val freeVars = terms.flatMap(freeVarsOf).toSet intersect quantVars
+        val errors = checkNegativeWildcards(terms, loc)
 
         // Check if any free variables are not positively bound.
-        ((freeVars -- posVars) map {
-          case unboundVar => SafetyError.IllegalNonPositivelyBoundVariable(unboundVar, loc)
-        }).toList
+        errors ++ ((freeVars -- posVars) map (unboundVar => SafetyError.makeIllegalNonPositivelyBoundVariableError(unboundVar, loc))).toList
     }
   }
 
+  /**
+    * @param loc the location of the atom containing the terms.
+    */
+  private def checkNegativeWildcards(terms: List[TypedAst.Pattern], loc: SourceLocation): List[CompilationError] = {
+    terms.flatMap(checkNegativeWildcards(_, loc))
+  }
+
+  /**
+    * @param loc the location of the atom containing the term.
+    */
+  @tailrec
+  private def checkNegativeWildcards(term: TypedAst.Pattern, loc: SourceLocation): List[CompilationError] = term match {
```

Comment:
I think we should just rename these to `visitPat` and `visitPats`. Thats the usual names.